packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -290,18 +291,45 @@ export const createBundle = async ({
                 } = await getProjectPkg()
                 const extendsTemplate = 'node_modules/' + ccExtensibility.extends
 
+                const pwaKitDeps = [
+                    '@salesforce/pwa-kit-dev',
+                    '@salesforce/pwa-kit-runtime',
+                    '@salesforce/pwa-kit-react-sdk'
+                ]
+
                 let cc_overrides: string[] = []
-                if (ccExtensibility.overridesDir) {
+                const packageVersions: {[key: string]: string} = {}
+
+                if (ccExtensibility.extends) {
+                    const extendsTemplateVersion =
+                        dependencies[ccExtensibility.extends] ||
+                        devDependencies[ccExtensibility.extends]
                     const overrides_files = await walkDir(
                         ccExtensibility.overridesDir,
                         ccExtensibility.overridesDir
                     )
                     cc_overrides = Array.from(overrides_files).filter((item) =>
                         existsSync(path.join(extendsTemplate, item))
                     )
+
+                    // If the project is using template extensbility, we must use 'npm info' to find the versions of pwa-kit packages
+                    try {
+                        const packageInfo = JSON.parse(
+                            execSync(
+                                `npm info ${ccExtensibility.extends}@${extendsTemplateVersion} --json`
+                            ).toString()
+                        )
+                        pwaKitDeps.forEach((dep) => {
+                            if (packageInfo.dependencies?.[dep]) {
+                                packageVersions[dep] = packageInfo.dependencies[dep]
+                            }
+                        })
+                    } catch (error) {
+                        console.error('Error:', error)
```

Comment:
```suggestion
                        console.warn(`Could not parse dependencies for ${ccExtensibility.extends}.`)
```
[minor] How do we want to handle an error creating the metadata (e.g. if someone tries to upload a bundle during an npm outage)? Feels like it probably shouldn't be blocking. So I'd say this is more of a warning than an error. And the end user probably cares more about the context of the error than what actually went wrong (e.g. `Could not parse dependencies for package.` is more useful than `Error: SyntaxError: Unexpected token < in JSON at position 0`).
(Side note: There's utils in the `/bin/pwa-kit-dev.js` for fancy logging - you could copy that pattern if you want, or just leave it plain.)

Reply:
Thinking more about error handling, how do we want to distinguish between "no package versions because of an error" from "no package versions because they're not used"? I'm thinking maybe we have a default value for the packages, `"unknown"`. Then, if the package is found, set it to the found version. If it's not found, set it to `"unused"`. If there's an error, set it to `"error"`, and maybe add an `errors: [...]` field to the bundle metadata with details.

Reply:
This makes a lot of sense, I will include these changes.

I think for now, having `unknown` if the packages are unused, or if there is an error will suffice. And set the version if the package is found.